### DISCUSSION TITLE
how to generate shared library

### DISCUSSION THREAD

**Discussion by tommy-mor** [tree-sitter/tree-sitter#1711] [Q&A] [answered]
I've made a new grammar using the cli following [this](https://tree-sitter.github.io/tree-sitter/creating-parsers#command-generate) tutorial. I want to load it into emacs-tree-sitter so I can start using it. 

[This](https://emacs-tree-sitter.github.io/languages/) page tells me that a languages grammar is usually defined in a ``shared library (.dylib, .so, .dll).``

The cli command ``generate`` does not produce any of those files. I feel like I am missing something. The cli also does not have a ``build`` command, and the docs don't say anything about what to do next.

Does it have to do with node-gyp? If I do ``node-gyp build`` I feel like I'm getting closer, but still no .dylib, .so, or .dll, just .node and .target??.

**Comment by ahelwer** [comment]
The `test` and `parse` commands will compile the library. I think `npm install` does it too.

**Reply by tommy-mor** [reply]
when it compiles, where does it put the shared library that emacs can understand?

**Comment by tommy-mor** [comment]
the answer was hidden in this file: https://github.com/emacs-tree-sitter/tree-sitter-langs/blob/0dd5e56e2f5646aa51ed0fc9eb869a8f7090228a/tree-sitter-langs-build.el#L340

**Comment by jirutka** [comment]
You can take inspiration from the [abuild-tree-sitter](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/tree-sitter/abuild-tree-sitter) script that we use for building tree-sitter packages in Alpine Linux.

**Reply by tommy-mor** [reply]
thanks, will look at this next time I need to solve this problem

**Comment by ahlinc** [comment, marked as answer]
[Now it's possible](https://github.com/tree-sitter/tree-sitter/issues/1868#issuecomment-1374694690) with `tree-sitter generate -b --libdir .` and you'll find the builded `*.so` file in your current directory.
